src/filesystem/file.ts

```diff
@@ -19,7 +19,7 @@ export class File extends FileSystem {
     throw new MissingFileException();
   }
 
-  public async write(data: string): Promise<void> {
+  public async write(data: Parameters<typeof fs.writeFile>["1"]): Promise<void> {
```

Comment:
I don't oppose to this typing, it actually makes more sense, but can you add like a comment with the current type? I see it looks something like `string | NodeJS.ArrayBufferView`

```suggestion
/**
  * arguments:
  * data: string | NodeJS.ArrayBufferView
**/
  public async write(data: Parameters<typeof fs.writeFile>["1"]): Promise<void> {
```

Reply:
Honestly, I completely forgot about developer experience 😅 
Definitely looks funky. At least when you start typing the method like `something.write(..` it will suggest you the list of type it accepts as 1st argument, which is much nicer than the monster I added. Example: 
<img width="330" alt="image" src="https://user-images.githubusercontent.com/6705361/199085692-a1234275-abb0-4738-8724-9c07afec9795.png">

Anyway, added a comment as you suggested.